frontend/public/components/dashboards-page/with-dashboard-resources.tsx

```diff
@@ -84,11 +114,17 @@ type WithDashboardResourcesProps = {
   [RESULTS_TYPE.URL]: ImmutableMap<string, any>;
 };
 
+export type WatchK8sResource = (resource: FirehoseResource) => void;
+export type StopWatchK8sResource = (resource: FirehoseResource) => void;
+
 export type DashboardItemProps = {
-  watchURL: WatchURL,
-  stopWatchURL: StopWatchURL,
-  watchPrometheus: WatchPrometheus,
-  stopWatchPrometheusQuery: StopWatchPrometheus,
-  urlResults: ImmutableMap<string, any>,
-  prometheusResults: ImmutableMap<string, any>,
-}
+  watchURL: WatchURL;
+  stopWatchURL: StopWatchURL;
+  watchPrometheus: WatchPrometheus;
+  stopWatchPrometheusQuery: StopWatchPrometheus;
+  urlResults: ImmutableMap<string, any>;
+  prometheusResults: ImmutableMap<string, any>;
+  watchK8sResource: WatchK8sResource;
+  stopWatchK8sResource: StopWatchK8sResource;
+  resources?: object;
```

Comment:
Is there a better type we can use? Is this a `FirehoseResource[]`?

Reply:
that is actually what Firehose passes down - i've added better type but Im not sure if it is already created somewhere so I could reuse it.